apps/lend/src/hooks/useSupplyTotalApr.ts

```diff
@@ -0,0 +1,91 @@
+import { useMemo } from 'react'
+import useStore from '@/store/useStore'
+
+import { INVALID_ADDRESS } from '@/constants'
+import { FORMAT_OPTIONS, formatNumber } from '@/ui/utils'
+
+function useSupplyTotalApr(rChainId: ChainId, rOwmId: string) {
+  const owmData = useStore((state) => state.markets.owmDatasMapper[rChainId]?.[rOwmId])
+  const marketRewardsResp = useStore((state) => state.markets.rewardsMapper[rChainId]?.[rOwmId])
+  const marketRatesResp = useStore((state) => state.markets.ratesMapper[rChainId]?.[rOwmId])
+
+  const { gauge } = owmData?.owm?.addresses ?? {}
+  const { error: rewardsError } = marketRewardsResp ?? {}
+  const { error: ratesError } = marketRatesResp ?? {}
+
+  const { totalApr, tooltipValues } = useMemo(
+    () => _getTotalAndTooltip(marketRewardsResp, marketRatesResp),
+    [marketRewardsResp, marketRatesResp]
+  )
+
+  return {
+    isReady: typeof marketRewardsResp !== 'undefined' && typeof marketRatesResp !== 'undefined',
+    isError: rewardsError || ratesError,
+    invalidGaugeAddress: typeof gauge !== 'undefined' && gauge === INVALID_ADDRESS,
+    totalApr,
+    tooltipValues,
+  }
+}
+
+function _getTotalApr(lendApr: number, crvBase: number, crvBoost: number, others: RewardOther[]) {
+  const othersTotal = (others ?? []).reduce((prev, curr) => {
+    prev += curr.apy
+    return prev
+  }, 0)
+
+  const min = (lendApr + crvBase + othersTotal).toString()
+  const max = (lendApr + crvBoost + othersTotal).toString()
+
+  return {
+    min,
+    max,
+    minMax:
+      min === max
+        ? formatNumber(min, FORMAT_OPTIONS.PERCENT)
+        : `${formatNumber(min, FORMAT_OPTIONS.PERCENT)} - ${formatNumber(max, FORMAT_OPTIONS.PERCENT)}`,
+  }
+}
+
+function _getTooltipValue(lendApr: number, lendApy: number, crvBase: number, crvBoost: number, others: RewardOther[]) {
+  return {
+    lendApr: formatNumber(lendApr, FORMAT_OPTIONS.PERCENT),
+    lendApy: `${formatNumber(lendApy, FORMAT_OPTIONS.PERCENT)} APY`,
+    crv: crvBase > 0 ? formatNumber(crvBase, FORMAT_OPTIONS.PERCENT) : '',
+    crvBoosted: crvBoost > 0 ? formatNumber(crvBoost, FORMAT_OPTIONS.PERCENT) : '',
+    incentives: others.map((o) => `${formatNumber(o.apy, FORMAT_OPTIONS.PERCENT)} ${o.symbol}`),
+    incentivesObj: others,
+  }
+}
+
+function _getTotalAndTooltip(marketRewardsResp: MarketRewards, marketRatesResp: MarketRates) {
+  let resp: {
+    totalApr: { min: string; max: string; minMax: string }
+    tooltipValues: {
+      lendApr: string
+      lendApy: string
+      crv: string
+      crvBoosted: string
+      incentives: string[]
+      incentivesObj: RewardOther[]
+    } | null
+  } = {
+    totalApr: { min: '', max: '', minMax: '' },
+    tooltipValues: null,
+  }
+
+  if (typeof marketRewardsResp === 'undefined' || typeof marketRatesResp === 'undefined') return resp
```

Comment:
The default value only gets used here, we could inline it for clarity
By returning the data directly we don't need to declare the types
```suggestion
  if (!marketRewardsResp || !marketRatesResp) {
    return {
      totalApr: { min: '', max: '', minMax: '' },
      tooltipValues: null,
    }
  }
```